pkg/antctl/raw/multicluster/create/clusterset.go

```diff
@@ -0,0 +1,141 @@
+`, "\n")
+
+func (o *clusterSetOptions) validateAndComplete() error {
+	if o.namespace == "" {
+		return fmt.Errorf("the Namespace cannot be empty")
+	}
+	if o.leaderClusterNamespace == "" {
+		o.leaderClusterNamespace = metav1.NamespaceDefault
+	}
+	if o.leaderCluster == "" {
+		return fmt.Errorf("the leader-cluster-id cannot be empty")
+	}
+	if o.secret == "" && o.memberClusters == nil {
+		return fmt.Errorf("the ServiceAccounts list is required in leader cluster, the secret is required in member cluster")
+	}
+	if o.memberClusters == nil {
+		return fmt.Errorf("the ServiceAccounts list cannot be empty")
+	}
+
+	return nil
+}
+
+func NewClusterSetCmd() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "clusterset",
+		Args:    cobra.MaximumNArgs(1),
+		Short:   "Create a ClusterSet in a leader or member cluster",
+		Long:    "Create a ClusterSet in a leader or member cluster",
+		Example: clusterSetExamples,
+		RunE:    clusterSetRunE,
+	}
+
+	o := &clusterSetOptions{}
+	clusterSetOpt = o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of the ClusterSet")
+	command.Flags().StringVarP(&o.leaderCluster, "leader-cluster-id", "", "", "Leader cluster of the ClusterSet")
+	command.Flags().StringVarP(&o.leaderClusterServer, "leader-server", "", "", "Leader cluster server address of the ClusterSet")
+	command.Flags().StringVarP(&o.leaderClusterNamespace, "leader-namespace", "", "", "Leader cluster Namespace")
+	command.Flags().StringToStringVarP(&o.memberClusters, "member-clusters", "", nil, "clusterID and ServiceAccount group of the member clusters(e.g. --member-clusters member1=sa1,member2=sa2)")
+	command.Flags().StringVarP(&o.secret, "secret", "", "", "Secret to access the leader cluster,it is required only when creating ClusterSet in member cluster")
+
+	return command
+}
+
+func clusterSetRunE(cmd *cobra.Command, args []string) error {
+	if err := clusterSetOpt.validateAndComplete(); err != nil {
+		return err
+	}
+	if len(args) != 1 {
+		return fmt.Errorf("exactly one NAME is required, got %d", len(args))
+	}
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+	clusterSetName := args[0]
+	clusterSet := &multiclusterv1alpha1.ClusterSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterSetName,
+			Namespace: clusterSetOpt.namespace,
+		},
+		Spec: multiclusterv1alpha1.ClusterSetSpec{
+			Leaders: []multiclusterv1alpha1.MemberCluster{
+				{
+					ClusterID: clusterSetOpt.leaderCluster,
+					Secret:    clusterSetOpt.secret,
+					Server:    fmt.Sprintf("https://%s", strings.Replace(clusterSetOpt.leaderClusterServer, "https://", "", 1)),
```

Comment:
sorry, I don't quite get this replacement, what's the purpose?

Reply:
Some users may use "127.0.0.1:6443" and other may use "https://127.0.0.1:6443", this replacement will add "https://" to the first one.